### PR TITLE
Cli Migration from Freshli

### PR DESCRIPTION
**Pull request by dan-hein**
### Goal
The primary goal of these changes is to take the Cli components from Freshli and migrate them into this repo. 

### Details
To start, the Program.cs file and OutputFormatter.cs file have been migrated along with their git histories and tests into this repo and into new corresponding projects and solutions. To complement and solidify this change, on the cli-breakout branch, Freshli has been converted into a class library, and the migrated files have been removed.

The README has also been updated to reflect the proper build, run, and test steps and output.

### Next Steps
I believe this is ready to merge, but if there are any necessary adjustments, please let me know and I'd be more than happy to correct those issues! I believe that once this branch is ready and merged we can continue on the necessary improvements for the Cli.

### CODE REVIEW

File: Freshli.Cli.Test/Freshli.Cli.Test.csproj
Comment (dfar-io) [inline Freshli.Cli.Test/Freshli.Cli.Test.csproj:19]: So for this, the changes I'm making for Code Coverage reporting via CodeClimate will require the use of the MSBuild variant of Coverlet: https://github.com/coverlet-coverage/coverlet#msbuild-integration-suffers-of-possible-known-issue

I'll be updating the Core project to use it, so we should update it here as well so we can get coverage across projects.

File: .gitmodules
Comment (dfar-io) [inline .gitmodules:2]: So I think this actually works (it pulls down the correct commit regardless).. but should be `branch`

**Review by dfar-io — CHANGES_REQUESTED** [review]
A few more changes, and thinking we should add a task to improve coverage *after* merging this in (no reason to hold up the overall process).

Code coverage of the CLI currently is:
* Line 42.44%
* Branch 50%
* Method 40%

Would like to see it get to around 80%.

File: .gitmodules
Comment (dan-hein) [reply .gitmodules:2]: That is definitely an issue, thanks!

**Comment by mrbiggred** [comment]
@dan-hein I removed all but @dfar-io as reviewers to prevent confusion on who was supposed to review this issue.  I want to avoid the problem where everyone assigned to a task thinks someone else is going to do it.

That said, if you or Dave want another reviewer feel free to reach out to that person.

**Review by dfar-io — CHANGES_REQUESTED** [review]
I might be leaving this comment twice? Just the coverlet.msbuild change and this is good to merge.

**Review by dfar-io — APPROVED** [review]
:+1:

**Pull request merged**
